TeamPorcupine/ProjectPorcupine
# Varying input/output (power/fluid) for components

This allows to define furniture with fluctuating power input - meaning, if there is power shortage, instead of whole grid going down for these machines, machines will work at decreased efficiency (depending on available power)
E.g. here 2 smelters with power requirement of 2 (needed power = 4) being hooked on grid that provides 2 power. Efficiency varies from 50 to 60%. 
![image](https://cloud.githubusercontent.com/assets/12616499/22308848/008985fe-e349-11e6-8fe2-c4ef9f67ad31.png)

CanFluctuate flags are currently applied only on Oxygen Generator and Smelter.1

- [x] Fix showing efficiency/electricity description when offline (now showing 100%)
- [x] Do not consume power when not working
- [x] Do not consume inventory when not needed

Changes:
- Requires with params is now RunConditions in components
- Power grid info now shows whether it's connected to grid with at least one producer (as Online/Offline is kind of confusing now with machines that have means to run but are not required)
- IPluggable now has more flags to support 'on demand' producers (fission reactor)

Testcases:
Power on demand
- RTGs are providing power to oxygen generators and pumps
- spawn power cell
 => power cell goes into stockpile (there is enough power in grid)
- remove wires from RTGs that are next to fission reactor (disconnect power from RTGs to grid where reactor is in)
 => power goes offline in pumps and generator
 => power cell is hauled to fission reactor and reactor is producing power
- add power cables back to RTGs (so they are connected again)
 => fission reactor should be idle again, (power is provided from RTGs)
- remove cables from RTGs again
 => fission reactor continues to spend fuel and provide power
 
Grid efficiency
- create RTG, create another smelter
- connect both smelters to one RTG, assign iron smelting, spawn raw_iron
 => see power efficiency fluctuate when smelters are smelting
